### PR TITLE
[Features] Added orders, order items, and transactions entities

### LINKED ISSUE #102: Initialize Transaction, Orders, and Order_item entity.

### PR DESCRIPTION
**Pull request by ZeroFairy**
resolved #102 

## Description
Added entitties:
- Orders
- Order Items
- Transactions

### CODE REVIEW

**Comment by vianneynara** [comment]
Hmm kenapa masih fail ya

File: src/main/java/dev/realtards/kuenyawz/entities/Account.java [hunk 1]
```diff
@@ -84,6 +82,9 @@ public boolean isEnabled() {
 		return UserDetails.super.isEnabled();
 	}
 
+	@OneToMany(mappedBy = "account", fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = true)
+	private Set<Transactions> transactions;
+
```
Comment (vianneynara) [inline src/main/java/dev/realtards/kuenyawz/entities/Account.java:87]: Set FetchType as EAGER

File: src/main/java/dev/realtards/kuenyawz/entities/OrderItems.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import lombok.*;
+import org.hibernate.annotations.CollectionId;
+
+import java.math.BigDecimal;
+
+@Entity
+@Getter
+@Setter
+//@EqualsAndHashCode(callSuper = true, exclude = "product")
+//@ToString(callSuper = true, exclude = "product")
```
Comment (vianneynara) [inline src/main/java/dev/realtards/kuenyawz/entities/OrderItems.java:15]: Delete unused annotations

File: src/main/java/dev/realtards/kuenyawz/entities/Orders.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import org.springframework.data.domain.Auditable;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+
+@Entity
+@Getter
+@Setter
+//@EqualsAndHashCode(callSuper = true, exclude = "product")
+//@ToString(callSuper = true, exclude = "product")
```
Comment (vianneynara) [inline src/main/java/dev/realtards/kuenyawz/entities/Orders.java:17]: Delete unused annotations

File: src/main/java/dev/realtards/kuenyawz/entities/Transactions.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@Entity
+@Getter
+@Setter
+//@EqualsAndHashCode(callSuper = true, exclude = "product")
+//@ToString(callSuper = true, exclude = "product")
```
Comment (vianneynara) [inline src/main/java/dev/realtards/kuenyawz/entities/Transactions.java:16]: Delete unused annotations

File: src/main/java/dev/realtards/kuenyawz/entities/Orders.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    @JoinColumn(name = "dp_reference_id", referencedColumnName = "reference_id")
+    private Transactions dpReferenceId;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "fp_reference_id", referencedColumnName = "reference_id")
+    private Transactions fpReferenceId;
+
+    public enum OrderStatus {
+        DEPOSIT, FULL_PAID
+    }
```
Comment (vianneynara) [inline src/main/java/dev/realtards/kuenyawz/entities/Orders.java:48]: Enum and inner classes must be placed on the botto  of the file

File: src/main/java/dev/realtards/kuenyawz/entities/Transactions.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    @Column(name = "finalized_at")
+    private LocalDateTime finalizedAt;
+
+    @ManyToOne(fetch = FetchType.EAGER)
+    @JoinColumn(name = "account_id", nullable = false)
+    private Account account;
+
+    public enum TransactionStatus {
+        DEPOSIT, PAID
+    }
```
Comment (vianneynara) [inline src/main/java/dev/realtards/kuenyawz/entities/Transactions.java:52]: PENDING EXPIRED SUCCESS CANCELLED

File: src/main/java/dev/realtards/kuenyawz/entities/Orders.java (hunk 5 above)
Comment (vianneynara) [reply src/main/java/dev/realtards/kuenyawz/entities/Orders.java]: WAITING_DOWN_PAYMENT("Waiting for down payment")
CONFIRMING("Waiting for confirmation from seller")
CONFIRMED("Confirmed by seller")
WAITING_SETTLEMENT"Waiting for settlement")
PROCESSING("Being processed")
DELIVERED("Order delivered")
CANCELLED("Order cancelled")

**Review by vianneynara — CHANGES_REQUESTED** [review]
Please

**Review by vianneynara — APPROVED** [review]

**Review by vianneynara — APPROVED** [review]

**Comment by vianneynara** [comment]
all clear

**Comment by ZeroFairy** [comment]
Thank you

**Pull request merged**